### DISCUSSION TITLE
Training a Network with Hidden layers

### DISCUSSION THREAD

**Discussion by DS-567** [BindsNET/bindsnet#596] [General]
Up until now I have been using BindsNet to model and train a network with no hidden layers. I am now trying to add a hidden layer between the input and output layer. I have a couple of questions if anybody can help:

1. Do all 3 layers have to have the same number of neurons, or can all 3 layers have different layers, 
    e.g. 8 input nodes --> 20 hidden LIF nodes --> 2 output LIF nodes.

2. How does BindsNet implement supervised learning with hidden layers? Previously I have been clamping the output neuron to spike for the correct class to match the input sample label. Can this training code be the same with just adding the hidden layer to the network, or do you have to train the hidden layer separately?

Thanks for helping!

**Comment by Hananel-Hazan** [comment]
Adding another layer to a network is simple as adding another line (add_connection) to your script. 
See the Dot Tracing [example](https://github.com/BindsNET/bindsnet/blob/master/examples/dotTracing/dot_tracing.py) or any of the examples in [breakout](https://github.com/BindsNET/bindsnet/tree/master/examples/breakout)

Each layer can be in different size or shape (i.e. 3D) and can have different learning scheme or none at all using the definition of the Connection object see [here](https://github.com/BindsNET/bindsnet/blob/eff0c8fcafa3013c6c6975d89462ede1e22d7fe1/examples/dotTracing/dot_tracing.py#L223)

**Reply by DS-567** [reply]
Thank you Hananel, it worked perfectly!
